src/Misc.ino

```diff
@@ -1737,14 +1746,45 @@ String parseTemplate(String &tmpString, byte lineSize)
                                   case 'S' :// Suffix Fill with n spaces: Sn
                                     if (valueJustLength > 1)
                                     {
-                                      if (isDigit(valueJust[1])) //Check Sn where n is between 0 and 9
+                                      if (isDigit(valueJust[1])) //Check n where n is between 0 and 9
                                       {
                                         int filler = valueJust[1] - value.length() - '0' ; //48
                                         for (byte f = 0; f < filler; f++)
                                           value += " ";
                                       }
                                     }
                                     break;
+                                  case 'L': //left part of the string
+                                    if (valueJustLength > 1)
+                                    {
+                                      if (isDigit(valueJust[1])) //Check n where n is between 0 and 9
+                                      {
+                                        value = value.substring(0,(int)valueJust[1]-'0');
+                                      }
+                                    }
+                                    break;
+                                  case 'R': //Right part of the string
+                                    if (valueJustLength > 1)
+                                    {
+                                      if (isDigit(valueJust[1])) //Check n where n is between 0 and 9
+                                      {
+                                        value = value.substring(value.length()-((int)valueJust[1]-'0'));
```

Comment:
What if the given digit is longer than the string length?
And what if it is '0'?